morph_tool/utils.py

```diff
@@ -24,10 +25,32 @@ def iter_morphology_files(folder, recursive=False, extensions=None):
 def neurondb_dataframe(filename: Path) -> pd.DataFrame:
     '''Returns a DataFrame: [name, layer, mtype]
 
+    If read from an XML, additional columns maybe be present
     Args:
-        filename: the neurondb.dat file
+        filename: the neurondb.(dat|xml) file
     '''
-    columns = ['name', 'layer', 'mtype']
-    df = pd.read_csv(filename, sep=r'\s+', names=columns, usecols=range(len(columns)))
-    df.layer = df.layer.astype('str')
+    if filename.suffix.lower() == '.dat':
+        columns = ['name', 'layer', 'mtype']
+        df = pd.read_csv(filename, sep=r'\s+', names=columns, usecols=range(len(columns)))
+        df.layer = df.layer.astype('str')
+    elif filename.suffix.lower() == '.xml':
+        with open(filename) as fd:
```

Comment:
In case of Path the recommended way is `with filename.open() as fd:`. See https://docs.python.org/3/library/pathlib.html#basic-use